docs/manuals/gui/extension.md

```diff
@@ -1,42 +1,137 @@
-# Adding custom visual elements
-
 Although Taipy GUI comes with a set of visual elements that lets users create comprehensive
-user interface, there are situations where applications may need to provide a very specific
-kind of element, with capabilities that one cannot find in Taipy out-of-the-box.
+user interfaces, there are situations where applications may need to provide a very specific
+kind of element with capabilities that one cannot find in Taipy GUI out-of-the-box.
+
+# Adding custom visual elements
 
-Taipy GUI lets developers add their own visual elements in order to address specific use
+Taipy GUI lets developers create and use custom visual elements to address specific use
 cases or integrate third-party Web components. One can expand the functionality
 offered by the base Taipy GUI package to create custom components that can be
 effortlessly used in pages and shared with the community.
 
-A custom visual element is made of two parts:
+Custom visual elements are grouped into *Element Libraries*, where each element is identified
+by its name. An element name must be unique in the context of its library.<br/>
+Each library also as a name. To insert a custom visual element in a page, you will use
+the full name of the element:
+
+- `<library_name>.<element_name>` in a Markdown page
+- `<library_name>:<element_name>` in an HTML page
+
+## Element Libraries and Custom Elements
+
+An Element Library  holds the information necessary to refer to and instantiate custom visual elements. Element libraries can hold several visual element descriptors, and can be packaged
+into a standalone Python package that a Taipy GUI application can import or that can be
+deployed.
+
+Here are important things you need to know about element libraries:
+
+- An element library is a Python object that derives from the `ElementLibrary^` class. 
+- Element libraries have a name (returned by `ElementLibrary.get_name()^`) that is
+  used to refer to visual elements in page definitions.
+- Element libraries hold the set of the elements it holds, as returned by
+  `ElementLibrary.get_elements()^`.
+
+You can create an element library by simply instantiating the `ElementLibrary^`, and override the methods that are required. The two mandatory methods to override are
+`(ElementLibrary.)get_name()^` and `(ElementLibrary.)get_elements()^`: 
+
+```py
+from taipy.gui.extension import ElementLibrary
+
+class MyCustomLibrary(ElementLibrary):
+    def get_name(self) -> str:
+        return "<mu_custom_library_name>"
+
+    def get_elements(self) -> dict:
+        return ({
+          "<custom_element_1>": Element(...),
```

Comment:
why the <> ?

Reply:
To show it is a place holder... I'll rephrase that.